tests/include/mocks/namespaceprovidermock.hpp

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_NAMESPACEPROVIDER_MOCK_HPP_
+#define AOS_NAMESPACEPROVIDER_MOCK_HPP_
+
+#include <gmock/gmock.h>
+
+#include "aos/sm/networkmanager.hpp"
+
+namespace aos::sm::networkmanager {
+
+/**
+ * Namespace provider mock.
+ */
+class NamespaceProviderMock : public NamespaceProviderItf {
```

Comment:
Put into networkmanagermock.hpp keep mocks in the corresponding to original header files.